### PR TITLE
Finished drivetrain

### PR DESCRIPTION
**Pull request by mqxs21**
I made the drivetrain code manually since i had to wait to long to generate it, also set up pathplanner autochooser and drivetrain in robotcontainer.

### CODE REVIEW

**Comment by flashrun24** [comment]
Can you please explain why you have so many weird vendor deps such as DogLog, Studica, URCL, and two pathplanners

**Review by Jolteous — APPROVED** [review]
While this isn't exactly like the conventional way we do swerve drive, it is usable. You may run into some issues with driving that can be worked out later, but generally this seems good. Reminds me of @flashrun24's code from 2023.

**Comment by Amrit-s-RIS** [comment]
> Can you please explain why you have so many weird vendor deps such as DogLog, Studica, URCL, and two pathplanners @Amrit-s-RIS

I downloaded all the vendor dependencies I could from the WPILib menu in case i would need to use them while coding. I didn't think there would be anything wrong with this, does it cause issues with the rest of the code?

File: src/main/java/frc/robot/subsystems/SwerveBase.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    rearRight.setSwerveState(states[3]);
+  }
+  public void setSwerveWithChassis(ChassisSpeeds chassis){
+    setSwerveState(new SwerveDriveKinematics().toSwerveModuleStates(chassis));
+  }
+  public ChassisSpeeds getChassisSpeedsFromController(double vx, double vy, double rx){
+    vx = Math.abs(vx)>0.4 ? vx : 0;
+    vy = Math.abs(vy)>0.4 ? vy : 0;
+    rx = Math.abs(rx)>0.4 ? rx : 0;
+    return ChassisSpeeds.fromFieldRelativeSpeeds(vx, vy, rx, gyro.getRotation2d());
```
Comment (wlmchen) [inline src/main/java/frc/robot/subsystems/SwerveBase.java:43]: `.4` as a deadband is quite a lot, you'd have to move the joystick almost halfway to get any movement. Usually we go with something lower. Deadband is only really necessary if the controller drifts, i.e has tiny readings (usually under `.1`) without being touched

**Review by wlmchen — CHANGES_REQUESTED** [review]

**Comment by wlmchen** [comment]
> > Can you please explain why you have so many weird vendor deps such as DogLog, Studica, URCL, and two pathplanners @Amrit-s-RIS
> 
> I downloaded all the vendor dependencies I could from the WPILib menu in case i would need to use them while coding. I didn't think there would be anything wrong with this, does it cause issues with the rest of the code?

If we aren't using it, it just adds initial build+fetch time and overhead(?), doesn't affect the code but still best to remove it. There's also [Namespace Pollution](https://stackoverflow.com/questions/29707809/does-namespace-pollution-in-java-or-c-sharp-exist-like-in-c) if you're curious.

**Review by MostOriginalIGN — CHANGES_REQUESTED** [review]
Looks great so far other than the two Pathplanners in vendordeps and the other random libraries. This should be cleaned to prevent overhead and I believe the two pathplanners may cause conflicts as the share the same package names.

Also, you could look at [CTRE's examples repo](https://github.com/CrossTheRoadElec/Phoenix6-Examples/tree/main/java/SwerveWithPathPlanner) where they have the generated swerve code already.  All you would need to do later is to generate TunerConstants. There no shame in copy pasting the CommandSwerveDrivetrain code as it's already generated anyways.

**Review by flashrun24 — CHANGES_REQUESTED** [review]
I don't have anything to add, just make the changes mentioned by William and win

**Comment by mqxs21** [comment]
i made the changes and cleaned up the vendordeps

**Review by flashrun24 — APPROVED** [review]
Although there is more to be done I think that this is good to merge for now :)

**Review by wlmchen — APPROVED** [review]

**Review by MostOriginalIGN — APPROVED** [review]

**Comment by Jolteous** [comment]
Go ahead and merge @mqxs21

**Pull request merged**